db/index.js

```diff
@@ -36,29 +37,246 @@ async function getAllProducts() {
 
 
 async function createProduct(product) {
-
   const { name, description, price, imageURL, inStock, category } = product
-  
   try {
-   
     const { rows: [ newProduct ] } = await client.query(`
       INSERT INTO products(name, description, price, imageURL, "inStock", category) 
       VALUES($1, $2, $3, $4, $5, $6) 
       RETURNING *;
     `, [name, description, price, imageURL, inStock, category]);
 
-       return newProduct;
+    return newProduct;
+  } catch (error) {
+    throw error;
+  }
+};
+
+
+async function createUser({firstName, lastName, email, imageurl, username, password, isAdmin}) {
+  const hashedPassword = await bcrypt.hash(password, SALT_COUNT);
+  try {
+      const { rows: [user] } = await client.query(`
+      INSERT INTO users("firstName", "lastName", email, imageurl, username, password, "isAdmin")
+      VALUES($1, $2, $3, $4, $5, $6, $7)
+      RETURNING id, username;
+      `, [firstName, lastName, email, imageurl, username, hashedPassword, isAdmin]);
+
+      delete user.password
+      return user;
+  } catch (error) {
+    throw error;
+  }    
+};
+
+
+async function getUser({username, password}) {
+  try {
+      const {rows: [user] } = await client.query(`
+      SELECT *
+      FROM users
+      WHERE username = $1;
+      `, [username]);
+      console.log('theUser:', user)
+      const isMatch = bcrypt.compareSync(password, user.password);
+      if(!isMatch) {
+          return;
+
+      } else {
+        delete user.password;
+        return user;
+      }    
+  } catch (error) {
+      throw error;
+  }
+};
+
+async function getAllUsers() {
+  try {
+      const { rows } = await client.query(`
+      SELECT *
+      FROM users;
+      `);
+
+      return rows;
+  } catch (error) {
+      throw error;
+  }
+};
+
+
+async function getUserById(id) {
+  try {
+      const { rows: [user] } = await client.query(`
+      SELECT *
+      FROM users
+      WHERE id = $1;
+      `, [id])
+      delete user.password;
+
+      return user;
+  } catch (error) {
+    throw error;
+  }
+};
+
+
+async function getUserByUsername(username) {
+  try {
+    const { rows: [user] } = await client.query(`
+      SELECT *
+      FROM users
+      WHERE username=$1
+    `, [username]);
+
+      return user;
+  } catch (error) {
+    throw error;
+  }
+};
+
+
+async function getOrderById(id) {
+  try {
+    const {rows: [order]} = await client.query(`
+      SELECT *
+      FROM orders
+      WHERE id=$1
+      `, [id]);
+
+      const { rows: productList }  = await client.query(`
```

Comment:
Try to see if we can abstract away this logic into its own function.  Something like `getProductsByOrderId(id)`